examples/systemcalls/systemcalls.c

```diff
@@ -57,10 +70,32 @@ bool do_exec(int count, ...)
  *   (first argument to execv), and use the remaining arguments
  *   as second argument to the execv() command.
  *
-*/
-
-    va_end(args);
-
+ 
+*/  
+    
+    
+    pid_t pid;
+    int status;
+    
+    pid = fork();
+    if(pid == -1)
+    	return -1; //failed to fork
+    else if(pid == 0){
+    	execv(command[0], command);
+    	exit(-1);
+    }
+    if(waitpid(pid, &status, 0) == -1)
+    	return false;
+    else if (WIFEXITED(status)){
+    	int exit_status = WEXITSTATUS(status); 
+    	if(exit_status == status)
```

Comment:
@prvbharadwaj1731 I'm not sure this is the correct check here, I think you want to look for WEXITSTATUS == 0.